massmigration/migrations.py

```diff
@@ -59,53 +90,62 @@ def get_backend(self):
         backend_class = import_string(self.backend_str)
         return backend_class()
 
-    def launch(self):
+    def launch(self, db_alias):
         """ Pass the migration to the backend to perform the data operation(s).
             This is what should be called by the web interface to trigger the migration.
         """
-        self.check_dependencies()
+        allowed_db_aliases = self.get_allowed_db_aliases()
+        if db_alias not in allowed_db_aliases:
+            raise CannotRunOnDB(
+                f"Migration {self.key} can't run on {db_alias}. "
+                f"The allowed DBs for this migration are {', '.join(allowed_db_aliases)}. "
+            )
+
+        self.check_dependencies(db_alias)
         backend = self.get_backend()
         method = getattr(backend, self.backend_method)
-        method(self)
+        method(self, db_alias)
         logger.info("Launched migration %s on backend %s", self.key, backend.__class__)
 
-    def can_be_started(self) -> bool:
-        return not MigrationRecord.objects.filter(key=self.key).exists()
+    def can_be_started(self, db_alias) -> bool:
+        return not MigrationRecord.objects.using(db_alias).filter(key=self.key).exists()
 
-    def mark_as_started(self) -> UUID:
+    def mark_as_started(self, db_alias) -> UUID:
         """ Mark the migration as started in the database. Return the attempt UUID. """
-        with get_transaction().atomic():
-            if not self.can_be_started():
+        with get_transaction().atomic(using=db_alias):
```

Comment:
```suggestion
        with get_transaction(db_alias).atomic(using=db_alias):
```
This will require a corresponding tweak to the `get_transaction` function.

Reply:
Great catch, committed with the change to the get_transaction